pkg/users/git.go

```diff
@@ -86,14 +87,15 @@ func (r *GitUserResolver) Resolve(user *gits.GitUser) (*jenkinsv1.User, error) {
 			}
 		}
 		new := r.GitUserToUser(gitUser)
-		id = gitUser.Login
+		lowerCaseGitLogin := strings.ToLower(gitUser.Login)
```

Comment:
I don't think this is going to cut it, at least not for all cases. Lower casing is part of the issue since we use this login as a Kuberenetes label value, but there are more restrictions which apply. ` naming.ToValidName(gitUser.Login)` probably works. @mrageh has done this already and he also had a test.

Right now @mrageh tries to further improve the code, since passing `selectUsers` as a function parameter is super confusing and fulfils no real purpose really.